core/src/main/java/org/elasticsearch/common/logging/log4j/LogConfigurator.java

```diff
@@ -84,6 +85,10 @@
             .immutableMap();
 
     public static void configure(Settings settings) {
+        configure(settings, true);
+    }
+
+    public static void configure(Settings settings, boolean resolveConfig) {
```

Comment:
Probably deserves javadoc saying that resolveConfig controls if the config file is read or not.